app/src/main/java/ch/epfl/sdp/peakar/camera/CameraUiView.java

```diff
@@ -83,11 +85,33 @@
     private Bitmap mountainMarkerVisible;
     private Bitmap mountainMarkerNotVisible;
 
-    //List that contains the POIPoints
-    private List<POIPoint> POIPoints;
     //Map that contains the labeled POIPoints
     private Map<POIPoint, Boolean> labeledPOIPoints;
 
+    private final SharedPreferences sharedPref;
+
+    private Boolean displayedToastMode;
+
+    private final SharedPreferences.OnSharedPreferenceChangeListener listenerPreferences =
+            (prefs, key) -> {
+                String displayMode = prefs.getString("display_pois_preference", "0");
+                boolean filterPOIs = prefs.getBoolean("filter_pois", true);
+
+                switch (displayMode){
+                    case "1":
+                        setPOIs(filterPOIs ? ComputePOIPoints.getFilteredPOIsInSight() : ComputePOIPoints.getPOIsInSight());
+                        checkIfLineOfSightAvailable();
+                        break;
+                    case "2":
+                        setPOIs(filterPOIs ? ComputePOIPoints.getFilteredPOIsOutOfSight() : ComputePOIPoints.getPOIsOutOfSight());
+                        checkIfLineOfSightAvailable();
+                        break;
+                    default:
+                        //case 0 and default
+                        setPOIs(filterPOIs ? ComputePOIPoints.getFilteredPOIs() : ComputePOIPoints.getPOIs());
```

Comment:
I suggest to put the case 0 tag above the default, so we know that the case 0 behaves like the default one. 
Also please use a meaningful constant instead of "0"
```suggestion
                    case "0":
                    default:
                        //case 0 and default
                        setPOIs(filterPOIs ? ComputePOIPoints.getFilteredPOIs() : ComputePOIPoints.getPOIs());
```